### ISSUE TITLE
mp4 rotation metadata ignored for some videos

### ISSUE THREAD

**Issue by rothfels** [google/ExoPlayer#8928]
I've followed several threads about incorrectly rotated videos (https://github.com/google/ExoPlayer/issues/7553, https://github.com/google/ExoPlayer/issues/3275, https://github.com/google/ExoPlayer/issues/91) but none seem to apply here.

I have two (mp4) videos, both recorded in portrait mode on the same phone, one after the other. Both files metadata show `Rotation: 90°`:

![image](https://user-images.githubusercontent.com/1095573/117831313-4a024c80-b229-11eb-87aa-aa0ba05e073a.png)

ExoPlayer plays one video with rotation applied, the other without. On other clients (iPhone, MacOS, etc) both videos are played with rotation correctly applied.

I have a small reproduction of the problem in this repo: https://github.com/rothfels/exoplayerbug

I was also able to reproduce the problem in your demo app by adding the two URLs in the project above to `assets/media.exolist.json`.

- ExoPlayer version: 2.13.3
- Android version: 11
- Android device: Pixel 3

**Comment by Samrobbo** [comment]
@kim-vde Can you take a look at this? I had an initial look at it, it appears our parsing of the tkhd atom is not returning a rotation degree, when in theory it should? Could be something incorrect with the media, but seems unusual.

**Comment by kim-vde** [comment]
The problem for the broken video is that the [transformation matrix](https://en.wikipedia.org/wiki/Transformation_matrix#Rotation) in the tkhd box is incorrect. The value of `a10` [here](https://github.com/google/ExoPlayer/blob/r2.14.0/library/extractor/src/main/java/com/google/android/exoplayer2/extractor/mp4/AtomParsers.java#L821) is equal to 65536 while it should be -65536. The rotation is therefore ignored.

The broken video has been recorded on a Pixel 3 running Android 11 using the Camera app, right? Did you modify the video by any way (e.g. using Google Photos or any other app)?

**Comment by kim-vde** [comment]
Note: I tried to reproduce on a Pixel 3 with Android 11 but couldn't.

**Comment by rothfels** [comment]
@kim-vde interesting, thanks for looking into this. The videos were actually both recorded on iOS; I only noticed they don't play properly on Android/ExoPlayer, whereas other video players I tested don't have a problem applying the rotation to both videos in spite of the different metadata.

I'm not sure why the transformation matrix would be different for the two files, but that doesn't sound like an issue for this project. Feel free to close, I can reopen with more information if I find more.
